internal/translation/teams/conversion.go

```diff
@@ -0,0 +1,131 @@
+package teams
+
+import (
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/pointer"
+
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+)
+
+type Team struct {
+	Roles    []string
+	TeamName string
+	TeamID   string
+}
+
+type AssignedTeam struct {
+	Usernames []string
+	TeamName  string
+	TeamID    string
+}
+
+type TeamUser struct {
+	Username string
+	UserID   string
+}
+
+func NewTeam(projTeamSpec *akov2.Team, teamID string) *Team {
+	if projTeamSpec == nil {
+		return nil
+	}
+
+	roles := make([]string, 0)
+	for _, role := range projTeamSpec.Roles {
+		roles = append(roles, string(role))
+	}
+
+	team := &Team{
+		Roles:  roles,
+		TeamID: teamID,
+	}
+
+	return team
+}
+
+func NewAssignedTeam(teamSpec *akov2.TeamSpec, teamID string) *AssignedTeam {
+	if teamSpec == nil {
+		return nil
+	}
+	usernames := make([]string, 0)
+	for _, username := range teamSpec.Usernames {
+		usernames = append(usernames, string(username))
+	}
+
+	team := &AssignedTeam{
+		TeamID:    teamID,
+		TeamName:  teamSpec.Name,
+		Usernames: usernames,
+	}
+
+	return team
+}
+
+func TeamFromAtlas(team *admin.TeamResponse) *Team {
+	if team == nil {
+		return nil
+	}
+	tm := &Team{
+		TeamID:   team.GetId(),
+		TeamName: team.GetName(),
```

Comment:
Are roles missing to be set here?

Reply:
Type admin.TeamResponse does not hold team roles. Would setting Roles to an empty slice be safer behaviour in this case @josvazg ?

Reply:
Empty slice and nil should be the same for a regular loop. On top of that, probably not being used.

Seems them this is only handling a Team name-id pair. I guess it does not make sense for us to have a separate type for just the name and id, without roles, and embed it in the team with assigned roles. In other words, if that is not used, I guess it is fine like this.